stablehlo/dialect/TypeInference.cpp

```diff
@@ -1818,56 +1818,57 @@ LogicalResult inferAfterAllOp(HloDialectInterface* dialect,
 }
 
 LogicalResult inferAllToAllOp(
-    std::optional<Location> location, Value operand, int64_t splitDimension,
-    int64_t concatDimension, int64_t splitCount,
+    std::optional<Location> location, ValueRange operands,
+    int64_t splitDimension, int64_t concatDimension, int64_t splitCount,
     DenseIntElementsAttr replicaGroups,
     SmallVectorImpl<ShapedTypeComponents>& inferredReturnShapes) {
   // all_to_all_c5, all_to_all_c7, all_to_all_i5
   if (failed(verifyReplicaGroups(location, replicaGroups,
                                  /*allGroupsMustHaveSameSize=*/true,
                                  /*useGlobalDeviceIds=*/false, splitCount)))
     return failure();
+  for (auto operand : operands) {
+    Type operandType = operand.getType();
+    auto operandRankedType = cast<RankedTensorType>(operandType);
```

Comment:
Nit: `auto operandType = cast<RankedTensorType>(operand.getType());` and update usages below.